### ISSUE TITLE
Design doc for moving user role management out of Okta

### ISSUE THREAD

**Issue by mehansen** [CDCgov/prime-simplereport#6956]
### Background
Result of a discussion in 11/9/23 eng sync
We currently manage user roles using groups in Okta - whether a user's Okta account is in a certain group determines whether they have a certain role at a certain org. This is a problem because we are storing information that's crucial to our app function in an external system. It also makes role management complex and hard to follow and causes us to maintain two sources of truth for our user list.

### Action Requested
Write up a design doc for how we can mitigate this risk and make role management easier to understand and modify.
How can we (1) store this data in our system, (2) copy the data over from Okta, and (3) use our system as the source of truth for authorization?
Is there a viable smaller effort just to keep a copy? General sentiment of eng team seemed to be that it wouldn't be worth the effort.
Reach out to Emma or Zedd with questions - they have historical context.

### Acceptance criteria
Design doc discussed in eng sync
Possible solutions with level of effort and benefits then presented to product as a possible work item

**Comment by emyl3** [comment]
Design doc: https://docs.google.com/document/d/1YhRh9UIWj1kWeT35tj7T5pnUjW6Rn7qyIVkiQL3cDFU/edit#heading=h.onis45iraztl

**Comment by emyl3** [comment]
Currently blocked as we need to get and wait on CDC response. See context here:
https://skylight-hq.slack.com/archives/C01LTSNKEPP/p1702324583336859

**Comment by mpbrown** [comment]
Unblocked following approval from CDC https://skylight-hq.slack.com/archives/C019PRYMR0Q/p1702930020679799
